File: csharp/Manager.cs
using System;
using System.Collections.Generic;

[Serializable]
public class Manager : FullTimeEmployee
{
    public List<Employee>? ManagedEmployees { get; set; }

    public Manager() : base() { }

    public Manager(string id, string firstName, string lastName, DateTime dob, string position, string department,
        double baseSalary, string benefits, List<Employee> managedEmployees)
        : base(id, firstName, lastName, dob, position, department, baseSalary, benefits)
    {
        ManagedEmployees = managedEmployees;
    }

    public override void ReportToManager()
    {
        if (Manager != null)
            Console.WriteLine($"{GetFullName()} (Manager) reports to manager {Manager.GetFullName()}");
        else
            Console.WriteLine($"{GetFullName()} (Manager) has no assigned manager.");
    }

    public void ApproveLeave(Employee e)
    {
        Console.WriteLine($"Manager {GetFullName()} approves leave for {e.GetFullName()}");
    }
}

File: java/Manager.java
import java.util.Date;
import java.util.List;

public class Manager extends FullTimeEmployee {
    public List<Employee> managedEmployees;

    public Manager(String id, String firstName, String lastName, Date dob, String position, String department,
            double baseSalary, String benefits, List<Employee> managedEmployees) throws InvalidDataException {
        super(id, firstName, lastName, dob, position, department, baseSalary, benefits);
        this.managedEmployees = managedEmployees;
    }

    @Override
    public void reportToManager() {
        if (manager != null) {
            System.out.println(getFullName() + " (Manager) reports to manager " + manager.getFullName());
        } else {
            System.out.println(getFullName() + " (Manager) has no assigned manager.");
        }
    }

    public void approveLeave(Employee e) {
        System.out.println("Manager " + getFullName() + " approves leave for " + e.getFullName());
    }
}